l20n/migrate/test_context_real_examples.py

```diff
@@ -0,0 +1,415 @@
+                            SOURCE(
+                                'aboutDownloads.dtd',
+                                'aboutDownloads.remove'
+                            )
+                        )
+                    )
+                ]
+            ),
+            FTL.Entity(
+                id=FTL.Identifier('pause-menuitem'),
+                traits=[
+                    FTL.Member(
+                        FTL.Keyword('label', 'html'),
+                        COPY(
+                            SOURCE(
+                                'aboutDownloads.dtd',
+                                'aboutDownloads.pause'
+                            )
+                        )
+                    )
+                ]
+            ),
+            FTL.Entity(
+                id=FTL.Identifier('resume-menuitem'),
+                traits=[
+                    FTL.Member(
+                        FTL.Keyword('label', 'html'),
+                        COPY(
+                            SOURCE(
+                                'aboutDownloads.dtd',
+                                'aboutDownloads.resume'
+                            )
+                        )
+                    )
+                ]
+            ),
+            FTL.Entity(
+                id=FTL.Identifier('cancel-menuitem'),
+                traits=[
+                    FTL.Member(
+                        FTL.Keyword('label', 'html'),
+                        COPY(
+                            SOURCE(
+                                'aboutDownloads.dtd',
+                                'aboutDownloads.cancel'
+                            )
+                        )
+                    )
+                ]
+            ),
+            FTL.Entity(
+                id=FTL.Identifier('remove-all-menuitem'),
+                traits=[
+                    FTL.Member(
+                        FTL.Keyword('label', 'html'),
+                        COPY(
+                            SOURCE(
+                                'aboutDownloads.dtd',
+                                'aboutDownloads.removeAll'
+                            )
+                        )
+                    )
+                ]
+            ),
+            FTL.Entity(
+                id=FTL.Identifier('delete-all-title'),
+                value=COPY(
+                    SOURCE(
+                        'aboutDownloads.properties',
+                        'downloadAction.deleteAll'
+                    )
+                )
+            ),
+            FTL.Entity(
+                id=FTL.Identifier('delete-all-message'),
+                value=PLURALS(
+                    SOURCE(
+                        'aboutDownloads.properties',
+                        'downloadMessage.deleteAll'
+                    ),
+                    FTL.ExternalArgument('num'),
+                    lambda var: REPLACE(
+                        var,
+                        {'#1': [FTL.ExternalArgument('num')]}
```

Comment:
As this is a function, it's not possible to scan for `SOURCE` nodes inside of it when `ctx.add_transforms` is called.

If we ever need to build nested select expressions (which is unlikely when migrating from DTD/properties), we can consider adding explicit dependency declarations for such transforms.  Something like the following, which uses a `REQUIRES` decorator:

```python
REQUIRES(
    SOURCE('path2', 'key2')               # explicit dependency
)(
    FTL.Entity(
        id=FTL.Identifier('foo'),
        value=PLURALS(
            SOURCE('path1', 'key1'),      # this will be autoscanne
            FTL.ExternalArgument('num1'),
            lambda var: PLURALS(
                SOURCE('path2', 'key2'),  # this won't be autoscanned
                FTL.ExternalArgument('num2'),
                lambda var: COPY(var)
            )
        )
    )
)
```